File: C#/Problems/Problems/Problem1.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Problems
{
    class Problem1 : Problem
    {

        int[] list;
        int k;
        bool exp;

        public Problem1(int[] list, int k, bool exp)
        {
            this.list = list;
            this.k = k;
            this.exp = exp;
        }

        public bool Calculate()
        {
            bool res = false;
            HashSet<int> hashSet = new HashSet<int>();
            foreach (int i in list)
            {
                if (hashSet.Contains(i))
                {
                    res = true;
                }
                else
                {
                    hashSet.Add(k - i);
                }
            }


            return res;
        }


    }
}

File: Scalas/src/main/scala/Problem1.scala
package problem

import scala.collection.mutable

class Problem1(list: List[Int], k: Int, exp: Boolean) extends Problem {


  override def Calculate(): Boolean = {
    var res = false
    var Set = new mutable.HashSet[Int]()
    for (elem <- list) {
      if (Set.contains(elem)){
        res = true
        return res==exp
      }else{
        Set.add(k-elem)
      }
    }



    return res==exp
  }
}
